src/test/java/htsjdk/samtools/util/IOUtilTest.java

```diff
@@ -114,6 +114,10 @@ public void testFileReadingAndWriting() throws IOException {
         String randomizedTestString = TEST_STRING + System.currentTimeMillis();
         for (String ext : TEST_FILE_EXTENSIONS) {
             File f = File.createTempFile(TEST_FILE_PREFIX, ext);
+            IOUtil.assertFileIsReadable(f);
+
+            CodeUtil.applyIfNotNull(f,IOUtil::assertFileIsReadable);
```

Comment:
This is a strange test for this method...  Why not give it its own test case in a new CodeUtilUnitTest class?